src/spikeinterface/generation/session_displacement_generator.py

```diff
@@ -0,0 +1,510 @@
+
+        firing_rates_hz = sorting_extra_outputs["firing_rates"][0]
+
+        if method == "by_amplitude_and_firing_rate":
+            neg_ampl = np.min(np.min(first_rec_templates, axis=2), axis=1)
+            assert np.all(neg_ampl < 0), "assumes all amplitudes are negative here."
+            score = firing_rates_hz * neg_ampl
+        else:
+            score = firing_rates_hz
+
+        order_idx = np.argsort(score)
+        ordered_rec_scalings = recording_amplitude_scalings["scalings"][rec_idx][order_idx, np.newaxis, np.newaxis]
+
+    elif method == "by_passed_order":
+
+        ordered_rec_scalings = recording_amplitude_scalings["scalings"][rec_idx][:, np.newaxis, np.newaxis]
+
+    else:
+        raise ValueError("`recording_amplitude_scalings['method']` not recognised.")
+
+    moved_templates *= ordered_rec_scalings
+
+
+def _check_generate_session_displacement_arguments(
+    num_units, recording_durations, recording_shifts, recording_amplitude_scalings, shift_units_outside_probe
+):
+    """
+    Function to check the input arguments related to recording
+    shift and scale parameters are the correct size.
+    """
+    expected_num_recs = len(recording_durations)
+
+    if len(recording_shifts) != expected_num_recs:
+        raise ValueError(
+            "`recording_shifts` and `recording_durations` must be "
+            "the same length, the number of recordings to generate."
+        )
+
+    if recording_amplitude_scalings and shift_units_outside_probe:
+        raise ValueError(
+            "At present, using `recording_amplitude_scalings` and "
+            "`shift_units_outside_probe` together is not supported."
+        )
+
+    shifts_are_2d = [len(shift) == 2 for shift in recording_shifts]
+    if not all(shifts_are_2d):
+        raise ValueError("Each record entry for `recording_shifts` must have two elements, the x and y shift.")
+
+    if recording_amplitude_scalings is not None:
+
+        keys = recording_amplitude_scalings.keys()
+        if not "method" in keys or not "scalings" in keys:
+            raise ValueError("`recording_amplitude_scalings` must be a dict with keys `method` and `scalings`.")
+
+        allowed_methods = ["by_passed_order", "by_amplitude_and_firing_rate", "by_firing_rate"]
+        if not recording_amplitude_scalings["method"] in allowed_methods:
+            raise ValueError(f"`recording_amplitude_scalings` must be one of {allowed_methods}")
+
+        rec_scalings = recording_amplitude_scalings["scalings"]
+        if not len(rec_scalings) == expected_num_recs:
+            raise ValueError("`recording_amplitude_scalings` 'scalings' must have one array per recording.")
+
+        if not all([len(scale) == num_units for scale in rec_scalings]):
+            raise ValueError(
+                "The entry for each recording in `recording_amplitude_scalings` "
+                "must have the same length as the number of units."
+            )
+
+
+def _update_kwargs_for_extended_units(
```

Comment:
this is a cool feature! in the single-session case, i have previously dealt with this by adding extra units that are off the probe to start. i'm wondering how this fits into the single session case... would this function be useful there too?